[BUG] Installation History - Truncation Error

### Component

Full Dashboard

### Performance Monitor Version

2.6.0

### SQL Server Version

Microsoft SQL Server 2022 (RTM-CU23) (KB5078297) - 16.0.4236.2 (X64)  	Jan 22 2026 17:50:56  	Copyright (C) 2022 Microsoft Corporation 	Enterprise Edition: Core-based Licensing (64-bit) on Windows Server 2022 Datacenter 10.0 <X64> (Build 20348: ) (Hypervisor)

### Windows Version

Windows 11 25H2

### Describe the Bug

Upon updating the PerformanceMonitor database, the update failed to log a record into the config.installation_history table due to string truncation.

### Steps to Reproduce

Update the PerformanceMonitor database.

### Expected Behavior

The update should insert a record into the config.installation_history table.

### Actual Behavior

Due to string truncation, no record was inserted into the config.installation_history table.

### Error Messages / Log Output

```text
Recording installation history...
[WARN] Could not record installation history: String or binary data would be truncated in table 'PerformanceMonitor.config.installation_history', column 'sql_server_version'. Truncated value: 'Microsoft SQL Server 2022 (RTM-CU23) (KB5078297) - 16.0.4236.2 (X64) 
	Jan 22 2026 17:50:56 
	Copyri'.
The statement has been terminated.
```

### Screenshots

_No response_

### Additional Context

I noticed the sql_server_version column was defined as nvarchar(255) not null.  The output of the instance's @@VERSION is 260 characters long.

I happened to still have the upgrade log up on screen.  Sharing the entirety of the log now.  The log does show the upgrade paths.

```
Checking for upgrades from v2.2.0 to v2.6.0...
Found 3 upgrade(s) to apply
Applying upgrade: 2.2.0-to-2.3.0
[OK]   01_widen_query_stats_columns.sql - Success
[OK]   02_widen_audit_columns.sql - Success
[OK]   03_add_growth_vlf_columns.sql - Success
[OK] Upgrade 2.2.0-to-2.3.0: 3 succeeded, 0 failed
Applying upgrade: 2.3.0-to-2.4.0
[OK]   01_add_growth_vlf_columns.sql - Success
[OK] Upgrade 2.3.0-to-2.4.0: 1 succeeded, 0 failed
Applying upgrade: 2.4.0-to-2.5.0
[OK]   01_widen_version_columns.sql - Success
[OK] Upgrade 2.4.0-to-2.5.0: 1 succeeded, 0 failed
[OK] Upgrades complete: 5 succeeded, 0 failed
Starting main installation...
Starting installation...
Executing 01_install_database.sql...
[OK] 01_install_database.sql - Success
Executing 02_create_tables.sql...
[OK] 02_create_tables.sql - Success
Executing 03_create_config_tables.sql...
[OK] 03_create_config_tables.sql - Success
Executing 04_create_schedule_table.sql...
[OK] 04_create_schedule_table.sql - Success
Executing 05_delta_framework.sql...
[OK] 05_delta_framework.sql - Success
Executing 06_ensure_collection_table.sql...
[OK] 06_ensure_collection_table.sql - Success
Executing 07_collect_wait_stats.sql...
[OK] 07_collect_wait_stats.sql - Success
Executing 08_collect_query_stats.sql...
[OK] 08_collect_query_stats.sql - Success
Executing 09_collect_query_store.sql...
[OK] 09_collect_query_store.sql - Success
Executing 10_collect_procedure_stats.sql...
[OK] 10_collect_procedure_stats.sql - Success
Executing 11_collect_query_snapshots.sql...
[OK] 11_collect_query_snapshots.sql - Success
Executing 12_query_snapshots_create_views.sql...
[OK] 12_query_snapshots_create_views.sql - Success
Executing 13_query_snapshots_retention.sql...
[OK] 13_query_snapshots_retention.sql - Success
Executing 14_collect_memory_stats.sql...
[OK] 14_collect_memory_stats.sql - Success
Executing 15_collect_memory_grant_stats.sql...
[OK] 15_collect_memory_grant_stats.sql - Success
Executing 16_collect_memory_clerks_stats.sql...
[OK] 16_collect_memory_clerks_stats.sql - Success
Executing 17_collect_cpu_scheduler_stats.sql...
[OK] 17_collect_cpu_scheduler_stats.sql - Success
Executing 18_collect_cpu_utilization_stats.sql...
[OK] 18_collect_cpu_utilization_stats.sql - Success
Executing 19_collect_perfmon_stats.sql...
[OK] 19_collect_perfmon_stats.sql - Success
Executing 20_collect_file_io_stats.sql...
[OK] 20_collect_file_io_stats.sql - Success
Executing 21_setup_blocked_process_xe.sql...
[OK] 21_setup_blocked_process_xe.sql - Success
Executing 22_collect_blocked_processes.sql...
[OK] 22_collect_blocked_processes.sql - Success
Executing 23_process_blocked_process_xml.sql...
[OK] 23_process_blocked_process_xml.sql - Success
Executing 24_collect_deadlock_xml.sql...
[OK] 24_collect_deadlock_xml.sql - Success
Executing 25_process_deadlock_xml.sql...
[OK] 25_process_deadlock_xml.sql - Success
Executing 26_blocking_deadlock_analyzer.sql...
[OK] 26_blocking_deadlock_analyzer.sql - Success
Executing 27_collect_ring_buffer_events.sql...
[OK] 27_collect_ring_buffer_events.sql - Success
Executing 28_collect_system_health_wrapper.sql...
[OK] 28_collect_system_health_wrapper.sql - Success
Executing 29_collect_default_trace.sql...
[OK] 29_collect_default_trace.sql - Success
Executing 30_collect_trace_management.sql...
[OK] 30_collect_trace_management.sql - Success
Executing 31_collect_trace_analysis.sql...
[OK] 31_collect_trace_analysis.sql - Success
Executing 32_collect_latch_stats.sql...
[OK] 32_collect_latch_stats.sql - Success
Executing 33_collect_spinlock_stats.sql...
[OK] 33_collect_spinlock_stats.sql - Success
Executing 34_collect_tempdb_stats.sql...
[OK] 34_collect_tempdb_stats.sql - Success
Executing 35_collect_plan_cache_stats.sql...
[OK] 35_collect_plan_cache_stats.sql - Success
Executing 36_collect_session_stats.sql...
[OK] 36_collect_session_stats.sql - Success
Executing 37_collect_waiting_tasks.sql...
[OK] 37_collect_waiting_tasks.sql - Success
Executing 38_collect_server_configuration.sql...
[OK] 38_collect_server_configuration.sql - Success
Executing 39_collect_database_configuration.sql...
[OK] 39_collect_database_configuration.sql - Success
Executing 41_schedule_management.sql...
[OK] 41_schedule_management.sql - Success
Executing 42_scheduled_master_collector.sql...
[OK] 42_scheduled_master_collector.sql - Success
Executing 43_data_retention.sql...
[OK] 43_data_retention.sql - Success
Executing 44_hung_job_monitor.sql...
[OK] 44_hung_job_monitor.sql - Success
Executing 45_create_agent_jobs.sql...
[OK] 45_create_agent_jobs.sql - Success
Executing 46_create_query_plan_views.sql...
[OK] 46_create_query_plan_views.sql - Success
Executing 47_create_reporting_views.sql...
[OK] 47_create_reporting_views.sql - Success
Executing 48_create_additional_reporting_views.sql...
[OK] 48_create_additional_reporting_views.sql - Success
Executing 49_health_parser_tables.sql...
[OK] 49_health_parser_tables.sql - Success
Executing 50_configuration_issues_analyzer.sql...
[OK] 50_configuration_issues_analyzer.sql - Success
Executing 51_collect_running_jobs.sql...
[OK] 51_collect_running_jobs.sql - Success
Executing 52_collect_database_size_stats.sql...
[OK] 52_collect_database_size_stats.sql - Success
Executing 53_collect_server_properties.sql...
[OK] 53_collect_server_properties.sql - Success
Executing 54_create_finops_views.sql...
[OK] 54_create_finops_views.sql - Success
Installing community dependencies...
Installing community dependencies...
Installing sp_WhoIsActive...
[WARN] Network error, retrying in 2s (1/3)...
[WARN] Network error, retrying in 4s (2/3)...
[ERROR] sp_WhoIsActive - Download failed: The SSL connection could not be established, see inner exception.
Installing DarlingData...
[WARN] Network error, retrying in 2s (1/3)...
[WARN] Network error, retrying in 4s (2/3)...
[ERROR] DarlingData - Download failed: The SSL connection could not be established, see inner exception.
Installing First Responder Kit...
[WARN] Network error, retrying in 2s (1/3)...
[WARN] Network error, retrying in 4s (2/3)...
[ERROR] First Responder Kit - Download failed: The SSL connection could not be established, see inner exception.
[WARN] Dependencies installed: 0/3
Executing 98_validate_installation.sql...
[OK] 98_validate_installation.sql - Success
Recording installation history...
[WARN] Could not record installation history: String or binary data would be truncated in table 'PerformanceMonitor.config.installation_history', column 'sql_server_version'. Truncated value: 'Microsoft SQL Server 2022 (RTM-CU23) (KB5078297) - 16.0.4236.2 (X64) 
	Jan 22 2026 17:50:56 
	Copyri'.
The statement has been terminated.
Running post-install validation...
Running initial collection to validate installation...
Executing master collector...
[OK] Master collector completed
[OK] Validation complete: 45 collectors succeeded, 0 failed
[OK] Validation: 45 collectors succeeded, 0 failed
Report saved: C:\Users\USERNAME\PerformanceMonitor_Install_SERVERNAME_INSTANCENAME_20260410_140740.txt
[OK] Installation completed successfully!
```

--- a/upgrades/2.4.0-to-2.5.0/01_widen_version_columns.sql
+++ b/upgrades/2.4.0-to-2.5.0/01_widen_version_columns.sql
@@ -3,6 +3,20 @@ Widen sql_server_version and sql_server_edition columns in config.installation_h
 Some @@VERSION strings exceed 255 characters (#712)
 */
 
+SET ANSI_NULLS ON;
+SET ANSI_PADDING ON;
+SET ANSI_WARNINGS ON;
+SET ARITHABORT ON;
+SET CONCAT_NULL_YIELDS_NULL ON;
+SET QUOTED_IDENTIFIER ON;
+SET NUMERIC_ROUNDABORT OFF;
+SET IMPLICIT_TRANSACTIONS OFF;
+SET STATISTICS TIME, IO OFF;
+GO
+
+USE PerformanceMonitor;
+GO
+
 IF EXISTS
 (
     SELECT

--- a/upgrades/README.md
+++ b/upgrades/README.md
@@ -36,30 +36,49 @@ The installer:
 
 ## Upgrade Script Guidelines
 
-1. **Always check before altering**: Use `IF NOT EXISTS` checks before adding columns/indexes
-2. **Be idempotent**: Scripts should be safe to run multiple times
-3. **Preserve data**: Never DROP tables with data (use ALTER/UPDATE instead)
-4. **Add comments**: Document why each change is being made
-5. **Test upgrade paths**: Test upgrading from each previous version
+1. **Start from `_template.sql`**: Copy the template for every new upgrade script — it has the required SET options and `USE PerformanceMonitor` that the installer depends on
+2. **Always check before altering**: Use `IF NOT EXISTS` / `IF EXISTS` checks before adding or modifying columns/indexes
+3. **Be idempotent**: Scripts should be safe to run multiple times
+4. **Preserve data**: Never DROP tables with data (use ALTER/UPDATE instead)
+5. **Add comments**: Document why each change is being made
+6. **Test upgrade paths**: Test upgrading from each previous version
 
 ## Example Upgrade Script
 
 ```sql
 /*
+Copyright 2026 Darling Data, LLC
+https://www.erikdarling.com/
+
 Upgrade from 1.0.0 to 1.1.0
 Adds execution context tracking to query_stats
 */
 
--- Add new column if it doesn't exist
-IF NOT EXISTS (
-    SELECT 1/0
+SET ANSI_NULLS ON;
+SET ANSI_PADDING ON;
+SET ANSI_WARNINGS ON;
+SET ARITHABORT ON;
+SET CONCAT_NULL_YIELDS_NULL ON;
+SET QUOTED_IDENTIFIER ON;
+SET NUMERIC_ROUNDABORT OFF;
+SET IMPLICIT_TRANSACTIONS OFF;
+SET STATISTICS TIME, IO OFF;
+GO
+
+USE PerformanceMonitor;
+GO
+
+IF NOT EXISTS
+(
+    SELECT
+        1/0
     FROM sys.columns
-    WHERE object_id = OBJECT_ID('collect.query_stats')
-    AND name = 'execution_context'
+    WHERE object_id = OBJECT_ID(N'collect.query_stats')
+    AND   name = N'execution_context'
 )
 BEGIN
     ALTER TABLE collect.query_stats
-    ADD execution_context nvarchar(128) NULL;
+        ADD execution_context nvarchar(128) NULL;
 
     PRINT 'Added execution_context column to collect.query_stats';
 END;

--- a/upgrades/_template.sql
+++ b/upgrades/_template.sql
@@ -1,0 +1,23 @@
+/*
+Copyright 2026 Darling Data, LLC
+https://www.erikdarling.com/
+
+Upgrade from X.Y.Z to X.Y.Z
+<describe what this upgrade does and why>
+*/
+
+SET ANSI_NULLS ON;
+SET ANSI_PADDING ON;
+SET ANSI_WARNINGS ON;
+SET ARITHABORT ON;
+SET CONCAT_NULL_YIELDS_NULL ON;
+SET QUOTED_IDENTIFIER ON;
+SET NUMERIC_ROUNDABORT OFF;
+SET IMPLICIT_TRANSACTIONS OFF;
+SET STATISTICS TIME, IO OFF;
+GO
+
+USE PerformanceMonitor;
+GO
+
+/* upgrade logic here — must be idempotent */

